lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -38,6 +38,8 @@ def << arg
 
     def collections= collections
       raise ArgumentError, "each collection must be a Hydra::PCDM::Collection" unless collections.all? { |c| collection? c }
+      raise ArgumentError, "you can't add a collection to itself" if collections.all? { |c| c.id == self.id }
+      raise ArgumentError, "you can't recusively nest collections" if collection_ancestor?(collections)
```

Comment:
Can we change the text here? I'd suggest.  "A collection can't be an ancestor of itself".